agent/app/agent.go

```diff
@@ -540,10 +540,8 @@ func (agent *ecsAgent) startAsyncRoutines(
 		StatsEngine:                   statsEngine,
 	}
 
-	if !agent.cfg.DisableMetrics {
-		// Start metrics session in a go routine
-		go tcshandler.StartMetricsSession(telemetrySessionParams)
-	}
+	// Start metrics session in a go routine
+	go tcshandler.StartMetricsSession(telemetrySessionParams)
```

Comment:
This is problematic because today, disabling metrics also means disabling the metrics collection engine that gets initialized (stats engine's init) from tcs handler. Customers might choose to do that because they want lighter cpu/memory footprint for the  ECS Agent. If `agent.cfg.DisableMetrics` is `false`, we should disable both metrics collection and reporting. I think with this change, we're only disabling the reporting pat. We should also disable metrics collection

Reply:
This has already been addressed. Disable the metrics won't affect the health check part.